frontend/src/App.vue

```diff
@@ -1,30 +1,96 @@
 <template>
-  <div>
-    <nav class="navbar bg-navbar navbar-dark shadow">
-      <div class="container-fluid">
-        <router-link class="navbar-brand" to="/">SUSE <i>aimaas</i></router-link>
-
+  <nav class="navbar bg-navbar navbar-dark navbar-expand-lg">
+    <div class="container-fluid">
+      <RouterLink class="navbar-brand" to="/">SUSE <i>aimaas</i></RouterLink>
+      <button class="navbar-toggler" type="button" data-bs-toggle="collapse"
+              data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent"
+              aria-expanded="false" aria-label="Toggle navigation">
+        <span class="navbar-toggler-icon"></span>
+      </button>
+      <div class="collapse navbar-collapse" id="navbarSupportedContent">
+        <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+          <SchemaList v-model="activeSchema" :as-dropdown="true" ref="schemalist"></SchemaList>
+        </ul>
       </div>
-    </nav>
-    <div class="container-fluid mt-2">
-      <router-view v-slot="{ Component }">
-        <keep-alive include="EntityList">
-          <component :is="Component"/>
-        </keep-alive>
-      </router-view>
     </div>
+  </nav>
+  <div class="container mt-2">
+    <keep-alive>
+      <router-view></router-view>
+    </keep-alive>
   </div>
 
 </template>
-
+<style>
+.eos-icons {
+  font-size: 120%;
+  line-height: 120%;
+  vertical-align: text-top;
+}
+</style>
 <script>
-import "bootstrap";
+import {computed} from "vue";
+import 'bootstrap/dist/js/bootstrap.min.js';
+import "eos-icons/dist/css/eos-icons.css";
 import "suse-bootstrap5-theme/dist/css/suse.css";
 
+import {api} from "@/api";
+import SchemaList from "@/components/SchemaList";
+
 export default {
   name: 'App',
-  components: {
-
-  }
+  metaInfo: {
+    title: 'SUSE aimaas'
+  },
+  components: {SchemaList,},
+  data: function () {
+    return {
+      api: api,
+      activeSchema: null
+    }
+  },
+  provide() {
+    return {
+      activeSchema: computed(() => this.activeSchema),
+      availableSchemas: computed(() => this.$refs.schemalist.schemas)
+    }
+  },
+  computed: {
+    availableSchemas() {
+      let _avail_schemas = {};
+      if (this.$refs.schemalist.schemas) {
+        for (let schema of this.$refs.schemalist.schemas) {
+          _avail_schemas[schema.slug] = schema;
+        }
+      }
+      return _avail_schemas
+    }
+  },
+  methods: {
+    getSchemaFromApi(schemaSlug) {
+        api.getSchema({slugOrId: schemaSlug}).then(schema => {
+          this.activeSchema = schema;
+        });
+    },
+    getSchemaFromRoute() {
+      let schemaSlug = this.$route.params.schemaSlug || null;
+      if (schemaSlug === undefined || schemaSlug === null) {
+        return null;
+      }
+      try {
+        // First, try to reuse data in storage
+        this.activeSchema = this.availableSchemas[schemaSlug];
+      } catch (e) {
+        true;
+      }
+      this.getSchemaFromApi(schemaSlug);
```

Comment:
> try to reuse data in storage

Not sure that it's a good idea, because schema could have changed🤔 Although they are not supposed to be changed frequently.

But even so, it doesn't look that `this.activeSchema = this.availableSchemas[schemaSlug];` has any effect since after all it gets to `this.getSchemaFromApi(schemaSlug);` which sets schema on its own🙃

Reply:
Good point. 

If I remember correctly, here we only care about the minimal schema information (name and slug to be exact). It might be safe to assume that these do not change often, as you already pointed out.

So, for the time being, I'd suggest to actually implement what I intended (get element from already fetched list first. If not possible, fetch from API).

Reply:
> I'd suggest to actually implement what I intended

okay👌